src/components/Notification.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { NotificationType } from '../types/NotificationType';
+import { useGlobalContext } from '../hooks/useGlobalContext';
+
+type Props = {
+  type: NotificationType;
+} & (
+  {
+    type: NotificationType.PostsWarning;
+    warningText: string
+  } | {
+    type: Exclude<NotificationType, NotificationType.PostsWarning>;
+    warningText?: string;
+  }
+);
+
+export const Notification: FC<Props> = ({ type, warningText }) => {
+  const { error } = useGlobalContext();
+
+  return (
+    <div
+      className={classNames('notification', {
+        'is-danger': error,
+        'is-warning': !error,
+      })}
+      data-cy={type}
```

Comment:
It's not clear why the data-cy attribute is being used here. If it's for testing purposes, consider using a tool like react-testing-library which encourages better testing practices.